Update SECURITY.md to use GH security advisory system (#257)
adam-fowler/mqtt-nio: SECURITY.md

## Supported Versions

Currently we support versions 2.x.x and later of MQTTNIO. These will receive security updates as and when needed.

## Reporting a Vulnerability

> If you believe you have found a security vulnerability in Hummingbird or any of its related repositories please do not post this in a public forum, do not create a GitHub Issue. Instead you should submit a security advisory via [GitHub](https://github.com/swift-server-community/mqtt-nio/security/advisories/new) with details of the issue.

Reviewer: ```suggestion
If you believe you have found a security vulnerability in MQTTNIO please do not post this in a public forum, do not create a GitHub Issue. Instead you should submit a security advisory via [GitHub](https://github.com/swift-server-community/mqtt-nio/security/advisories/new) with details of the issue.
```
😅

Author: I never copy and pasted that 😜
